library/lib.ml

```diff
@@ -432,6 +436,8 @@ module SynterpActions : LibActions with type summary = Summary.Synterp.frozen =
     let lib_synterp_stk = List.map (fun (node,_) -> drop_node node, []) st.lib_stk in
     { st with lib_stk = lib_synterp_stk }
 
+  let declare_info info = library_info := !library_info @ info
```

Comment:
The main action is done in Synterp only. Is that correct?

Reply:
that's also my understanding bu I'm no expert here